includes/express-checkout/class-wc-payments-express-checkout-button-display-handler.php

```diff
@@ -150,14 +150,31 @@ public function display_express_checkout_buttons() {
 			} else {
 				$this->payment_request_button_handler->display_payment_request_button_html();
 			}
+
+			if ( ! is_cart() ) {
+				?>
+				<wc-order-attribution-inputs id="wcpay-express-checkout__order-attribution-inputs"></wc-order-attribution-inputs>
+				<?php
+			} else {
+				add_action( 'woocommerce_after_cart', [ $this, 'add_order_attribution_inputs' ], 1 );
+			}
```

Comment:
```suggestion
			if ( is_cart() ) {
				add_action( 'woocommerce_after_cart', [ $this, 'add_order_attribution_inputs' ], 1 );
			} else {
				?>
				<wc-order-attribution-inputs id="wcpay-express-checkout__order-attribution-inputs"></wc-order-attribution-inputs>
				<?php
			}
```

This is a very pedantic suggestion, so feel free to ignore it 😅, but I feel like this is a bit clearer.

Also, while I'm at it with the pedantic suggestions, could we not do something like the below to avoid one line of duplication? 🤔 

 ```suggestion
			if ( is_cart() ) {
				add_action( 'woocommerce_after_cart', [ $this, 'add_order_attribution_inputs' ], 1 );
			} else {
				$this->add_order_attribution_inputs();
			}
```

Reply:
>However, from both the blocks cart and blocks checkout I was still able to produce an "Unknown" origin on my completed orders. 😢 Is that expected, @mdmoore, or would you consider this outside the scope of this issue, since it will probably require a separate fix? 🤔

Just noting that we discussed this earlier and the [PRB tokenized cart on block-based cart and checkout](https://github.com/Automattic/woocommerce-payments/issues/8527) hasn't been implemented yet. If order attributes  isn't covered with that implementation, we could create another issue to handle it.

I like the suggestions above. Changes in [7069fc0](https://github.com/Automattic/woocommerce-payments/pull/9038/commits/7069fc0b6e408367b0a347d9aa56e4bc66fe9edd)